File: 01_projects/code/java-playwright/cbaps-dawms/src/test/java/com/cbaps/pages/StatusTrackerPage.java
package com.cbaps.pages;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.microsoft.playwright.Locator;
import com.microsoft.playwright.Page;
import com.cbaps.library.PlaywrightManager;

/**
 * StatusTrackerPage: Final status validation page
 * Displays current requisition status after routing
 */
public class StatusTrackerPage {

	private static final Logger log = LoggerFactory.getLogger(StatusTrackerPage.class);
	private final Page page;
	private final PlaywrightManager pwm;

	private final Locator statusBadge;

	public StatusTrackerPage(Page page, PlaywrightManager pwm) {
		this.page = page;
		this.pwm = pwm;

		// Initialize locator
		this.statusBadge = page.locator("#reqStatus");

		// Stability anchor: wait for status to be visible
		pwm.waitVisible("#reqStatus");
		log.info("Status Tracker Page loaded successfully.");
	}

	/**
	 * Get current requisition status
	 * @return Status text
	 */
	public String getStatus() {
		String status = statusBadge.textContent();
		log.info("Retrieved status: " + status);
		return status;
	}
}

File: 01_projects/code/python-playwright/cbaps-dawms/basic-pom/cbaps/pages/status_tracker_page.py
"""
StatusTrackerPage: Final status validation page
Displays current requisition status after routing
"""

import logging
from playwright.sync_api import Page, Locator
from cbaps.library.playwright_manager import PlaywrightManager

logger = logging.getLogger(__name__)


class StatusTrackerPage:
    """
    StatusTrackerPage displays final requisition status.
    This is the Python equivalent of the Java StatusTrackerPage.
    """
    
    def __init__(self, page: Page, pwm: PlaywrightManager):
        """
        Initialize StatusTrackerPage
        
        Args:
            page: Playwright Page object
            pwm: PlaywrightManager instance
        """
        self.page = page
        self.pwm = pwm
        
        # Locators
        self.status_badge: Locator = page.locator("#reqStatus")
        
        # Stability anchor: wait for status to be visible
        pwm.wait_visible("#reqStatus")
        logger.info("Status Tracker Page loaded successfully.")
    
    def get_status(self) -> str:
        """
        Get current requisition status
        
        Returns:
            Status text
        """
        status = self.status_badge.text_content()
        logger.info(f"Retrieved status: {status}")
        return status
